iothub/service/tests/DigitalTwinsClientTests.cs

```diff
@@ -0,0 +1,471 @@
+                mockHttpRequestFactory,
+                s_retryHandler);
+
+            // act
+            Func<Task> act = async () => await digitalTwinsClient.GetAsync<string>(digitalTwinId);
+
+            // assert
+            await act.Should().ThrowAsync<IotHubServiceException>();
+        }
+
+        [TestMethod]
+        public async Task DigitalTwinsClient_UpdateAsync()
+        {
+            // arrange
+            string digitalTwinId = "foo";
+            var digitalTwin = new BasicDigitalTwin
+            {
+                Id = digitalTwinId,
+            };
+            var contents = new Dictionary<string, object>
+            {
+                { "temperature", 8 }
+            };
+            var jsonPatch = new JsonPatchDocument();
+            jsonPatch.AppendAdd("bar", contents);
+
+            var mockCredentialProvider = new Mock<IotHubConnectionProperties>();
+            mockCredentialProvider
+                .Setup(getCredential => getCredential.GetAuthorizationHeader())
+                .Returns(s_validMockAuthenticationHeaderValue);
+            var mockHttpRequestFactory = new HttpRequestMessageFactory(s_httpUri, "");
+
+            using var mockHttpResponse = new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.Accepted,
+                Content = HttpMessageHelper.SerializePayload(digitalTwin),
+            };
+            mockHttpResponse.Headers.Add("ETag", "\"AAAAAAAAAAE=\"");
+            mockHttpResponse.Headers.Add("Location", "foo");
+            
+            var mockHttpClient = new Mock<HttpClient>();
+            mockHttpClient
+                .Setup(restOp => restOp.SendAsync(It.IsAny<HttpRequestMessage>(), It.IsAny<CancellationToken>()))
+                .ReturnsAsync(mockHttpResponse);
+
+            var digitalTwinsClient = new DigitalTwinsClient(
+                HostName,
+                mockCredentialProvider.Object,
+                mockHttpClient.Object,
+                mockHttpRequestFactory,
+                s_retryHandler);
+
+            // act
+            DigitalTwinUpdateResponse response = await digitalTwinsClient.UpdateAsync(digitalTwinId, jsonPatch.ToString());
+
+            // assert
+            response.Should().BeOfType<DigitalTwinUpdateResponse>();
+        }
+
+        [TestMethod]
+        [DataRow(null, "foo")]
+        [DataRow("bar", null)]
+        [DataRow(null, null)]
+        public async Task DigitalTwinsClient_UpdateAsync_NullParamater_Throws(string digitalTwinId, string jsonPatch)
+        {
+            // arrange
+            using var serviceClient = new IotHubServiceClient(s_connectionString);
+
+            // act
+            Func<Task> act = async () => await serviceClient.DigitalTwins.UpdateAsync(digitalTwinId, jsonPatch);
+
+            await act.Should().ThrowAsync<ArgumentNullException>();
+        }
+
+        [TestMethod]
+        public async Task DigitalTwinsClient_UpdateAsync_HttpException()
```

Comment:
Could we rename this to something like "DigitalTwinsClient_UpdateAsync_OnHttpRequestException_ThrowsIotHubServiceException"?